### PR TITLE
Test/user class test suite

### LINKED ISSUE #6: Create User test suite

### PR DESCRIPTION
**Pull request by PaulTimothyChambers**
**What (if any) features are you implementing?**
- N/A
**What (if anything) did you refactor?**
- N/A

**Were there any issues that arose?**
- N/A

**Any other comments, questions, or concerns?**
- N/A

**Is there anything you need from your team?**
- A user class file and feedback (for honing the tests)

### CODE REVIEW

File: test/User-test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+const assert = require('chai').assert;
```
Comment (jfargnoli01) [inline test/User-test.js]: change to import statements bs webpack?

File: test/User-test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  it('should be able to remove a recipe from the user\'s weekly meal plan', function() {
+    user.addWeeklyRecipe(recipeOne);
+    user.addWeeklyRecipe(recipeTwo);
+    user.addWeeklyRecipe(recipeThree);
+
+    user.removeWeeklyRecipe(recipeTwo);
+    expect(user.favoriteRecipes).to.deep.equal([recipeOne, recipeThree]);
+  });
+});
```
Comment (jfargnoli01) [inline test/User-test.js]: remove semicolons that separate blocks  and end of class

**Review by jfargnoli01 — DISMISSED** [review]
Looks great - just a couple of minor adjustments needed before merging!

File: test/User-test.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+describe('User', function() {
+  let user;
+  let recipeOne;
+  let recipeTwo;
+  let recipeThree;
+
+  beforeEach(() => {
+    user = new User(users.name, users.id, users.pantry);
+    recipeOne = new Recipe(
```
Comment (nzambonivergara) [inline test/User-test.js]: The Recipe class takes in an object with the data that you are passing, so adding curly brackets is needed here I think.

Comment (nzambonivergara) [reply test/User-test.js]: Same applies for the instance of User, as the data will give us an object.

File: test/User-test.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  it('should have a name', function() {
+    expect(user.name).to.equal('Saige O\'Kon');
+  });
+
+  it('should have an ID', function() {
+    expect(user.id).to.equal(1);
+  });
+
+  it('should have a pantry of ingredients', function() {
+    expect(user.pantry).to.equal(users.pantry);
```
Comment (nzambonivergara) [inline test/User-test.js]: Should this be to.equal(users[0].pantry)?

File: test/User-test.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    user.removeFavorite(recipeTwo);
+    expect(user.favoriteRecipes).to.deep.equal([recipeOne, recipeThree]);
+  });
+
+  it('should be able to filter favorites by recipe tag', function() {
+    user.addFavorite(recipeOne);
+    user.addFavorite(recipeTwo);
+    user.addFavorite(recipeThree);
+
+    user.filterFavoriteRecipesByTags([recipeOne, recipeTwo, recipeThree]);
```
Comment (nzambonivergara) [inline test/User-test.js]: I think the argument passed here should be the tag(s) that the user will choose?

File: test/User-test.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    expect(user.filteredFavoritesByTag).to.deep.equal([recipeOne, recipeTwo]);
+  });
+
+  it('should be able to filter favorites by ingredient', function() {
+    user.addFavorite(recipeOne);
+    user.addFavorite(recipeTwo);
+    user.addFavorite(recipeThree);
+
+    user.filterFavoriteRecipesByIngredient([recipeOne, recipeTwo, recipeThree]);
```
Comment (nzambonivergara) [inline test/User-test.js]: same as line 75 but instead of tag(s), ingredient(s)?

File: test/User-test.js (hunk 4 above)
Comment (PaulTimothyChambers) [reply test/User-test.js]: that is correct! I have changed it

File: test/User-test.js (hunk 5 above)
Comment (PaulTimothyChambers) [reply test/User-test.js]: 👍

File: test/User-test.js (hunk 6 above)
Comment (PaulTimothyChambers) [reply test/User-test.js]: 👍

File: test/User-test.js (hunk 3 above)
Comment (PaulTimothyChambers) [reply test/User-test.js]: that's a much better idea! changing it now

**Review by jfargnoli01 — COMMENTED** [review]
Looks great, Paul! Believe I should be set to start the User class!

**Pull request merged**